### PR TITLE
Fix tests in NonStandardErpFederationsTest

### PR DESCRIPTION
**Pull request by marcos-iov**

### CODE REVIEW

File: rskj-core/src/test/java/co/rsk/peg/federation/NonStandardErpFederationsTest.java [hunk 1]
```diff
@@ -611,7 +605,7 @@ void createErpFederation_mainnet_constants_after_RSKIP293() {
     @Test
     void getRedeemScript_before_RSKIP_284_testnet() {
         networkParameters = NetworkParameters.fromID(NetworkParameters.ID_TESTNET);
-        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        activations = mock(ActivationConfig.ForBlock.class);
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/NonStandardErpFederationsTest.java:608]: do we need to mock this again? is not already done in the `setup()` ?

File: rskj-core/src/test/java/co/rsk/peg/federation/NonStandardErpFederationsTest.java [hunk 2]
```diff
@@ -1005,7 +997,7 @@ private void validateErpRedeemScript(
         assertEquals(ScriptOpCodes.getOpCode(String.valueOf(n)), script[index++]);
 
         assertEquals(ScriptOpCodes.OP_ENDIF, script[index++]);
-        assertEquals(Integer.valueOf(ScriptOpCodes.OP_CHECKMULTISIG).byteValue(), script[index++]);
+        assertEquals(Integer.valueOf(ScriptOpCodes.OP_CHECKMULTISIG).byteValue(), script[index]);
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/NonStandardErpFederationsTest.java:1000]: why this change?

File: rskj-core/src/test/java/co/rsk/peg/federation/NonStandardErpFederationsTest.java [hunk 3]
```diff
@@ -61,7 +55,7 @@ class NonStandardErpFederationsTest {
 
     @BeforeEach
     void setup() {
-        BridgeConstants bridgeConstants = BridgeMainNetConstants.getInstance();
+        FederationConstants federationConstants = FederationMainNetConstants.getInstance();
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/NonStandardErpFederationsTest.java:58]: would it make sense to call them `federationMainnetConstants`?

File: rskj-core/src/test/java/co/rsk/peg/federation/NonStandardErpFederationsTest.java [hunk 4]
```diff
@@ -465,10 +459,10 @@ void createdRedeemScriptProgramFromNonStandardErpBuilder_withRealValues_equalsRe
     @Test
     void createdFederationInfo_withRealValues_equalsExistingFederationInfo_testnet() {
         // values from last real non-standard erp fed in testnet
-        BridgeConstants bridgeTestNetConstants = BridgeTestNetConstants.getInstance();
-        networkParameters = bridgeTestNetConstants.getBtcParams();
-        emergencyKeys = bridgeTestNetConstants.getErpFedPubKeysList();
-        activationDelayValue = bridgeTestNetConstants.getErpFedActivationDelay();
+        FederationConstants federationTestNetConstants = FederationTestNetConstants.getInstance();
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/NonStandardErpFederationsTest.java:462]: and have `federationTestnetConstants` at a class level?

File: rskj-core/src/test/java/co/rsk/peg/federation/NonStandardErpFederationsTest.java (hunk 2 above)
Comment (marcos-iov) [reply rskj-core/src/test/java/co/rsk/peg/federation/NonStandardErpFederationsTest.java:1000]: No need to increase the value of `index` at that point. It's the last time it's used

File: rskj-core/src/test/java/co/rsk/peg/federation/NonStandardErpFederationsTest.java (hunk 1 above)
Comment (marcos-iov) [reply rskj-core/src/test/java/co/rsk/peg/federation/NonStandardErpFederationsTest.java:608]: It's basically overriding the mock created in the setup

File: rskj-core/src/test/java/co/rsk/peg/federation/NonStandardErpFederationsTest.java (hunk 4 above)
Comment (marcos-iov) [reply rskj-core/src/test/java/co/rsk/peg/federation/NonStandardErpFederationsTest.java:462]: I tried doing something similar but everything is a bit tangled, wouldn't get into those changes right now

File: rskj-core/src/test/java/co/rsk/peg/federation/NonStandardErpFederationsTest.java (hunk 3 above)
Comment (marcos-iov) [reply rskj-core/src/test/java/co/rsk/peg/federation/NonStandardErpFederationsTest.java]: updated

File: rskj-core/src/test/java/co/rsk/peg/federation/NonStandardErpFederationsTest.java (hunk 1 above)
Comment (julia-zack) [reply rskj-core/src/test/java/co/rsk/peg/federation/NonStandardErpFederationsTest.java:608]: but why mocking it again?

Comment (marcos-iov) [reply rskj-core/src/test/java/co/rsk/peg/federation/NonStandardErpFederationsTest.java:608]: To override the original object

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderTests.java [hunk 5]
```diff
@@ -243,7 +242,7 @@ private void testGetOldFederation(
     }
 
     @Test
-    void saveNewFederation_before_RSKIP123_should_allow_to_save_any_fed_type() throws IOException {
+    void saveNewFederation_before_RSKIP123_should_allow_to_save_any_fed_type() {
         testSaveNewFederation(
```
Comment (nathanieliov) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderTests.java:246]: Why not just create a parametrized test?

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderTests.java [hunk 6]
```diff
@@ -261,7 +260,7 @@ void saveNewFederation_before_RSKIP123_should_allow_to_save_any_fed_type() throw
     }
 
     @Test
-    void saveNewFederation_after_RSKIP123_should_save_standard_multisig_fed_format() throws IOException {
+    void saveNewFederation_after_RSKIP123_should_save_standard_multisig_fed_format() {
```
Comment (nathanieliov) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderTests.java:263]: I think this can be parametrized.

File: rskj-core/src/test/java/co/rsk/peg/federation/NonStandardErpFederationsTest.java (hunk 1 above)
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/federation/NonStandardErpFederationsTest.java:608]: Since the setup is doing this `when(activations.isActive(ConsensusRule.RSKIP201)).thenReturn(true);`, it needs to be overridden if you want to have other specific RSKIPs mocked.

File: rskj-core/src/test/java/co/rsk/peg/federation/NonStandardErpFederationsTest.java (hunk 2 above)
Comment (nathanieliov) [inline rskj-core/src/test/java/co/rsk/peg/federation/NonStandardErpFederationsTest.java:1000]: Why didn't this change break the test since you didn't update anything else here?

File: rskj-core/src/test/java/co/rsk/peg/federation/NonStandardErpFederationsTest.java (hunk 1 above)
Comment (j-zack) [reply rskj-core/src/test/java/co/rsk/peg/federation/NonStandardErpFederationsTest.java:608]: but there's no need to mock the whole object again. We just have to mock the response to the rskip284.

Comment (julia-zack) [reply rskj-core/src/test/java/co/rsk/peg/federation/NonStandardErpFederationsTest.java:608]: but there's no need to mock the whole object again right? We just have to mock the response to the rskip284

File: rskj-core/src/test/java/co/rsk/peg/federation/NonStandardErpFederationsTest.java (hunk 2 above)
Comment (julia-zack) [reply rskj-core/src/test/java/co/rsk/peg/federation/NonStandardErpFederationsTest.java:1000]: is using the same `index` value, just not incrementing it after

Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/federation/NonStandardErpFederationsTest.java:1000]: Got it

File: rskj-core/src/test/java/co/rsk/peg/federation/NonStandardErpFederationsTest.java (hunk 1 above)
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/federation/NonStandardErpFederationsTest.java:608]: The thing is, the old mock object may have mock things that you are not interested in.

Comment (julia-zack) [reply rskj-core/src/test/java/co/rsk/peg/federation/NonStandardErpFederationsTest.java:608]: we should have the rskip201 activated in all the tests (thats why is being activated in the setup). We are not mocking the whole object again in any of the other tests, just adding the mocked responses we need.

Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/federation/NonStandardErpFederationsTest.java:608]: If that's the case, you are right, no need for creating a new instance.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderTests.java (hunk 5 above)
Comment (marcos-iov) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderTests.java:246]: That's another story, the intention here is to simply fix the failing tests

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderTests.java (hunk 6 above)
Comment (marcos-iov) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderTests.java:263]: Same

File: rskj-core/src/test/java/co/rsk/peg/federation/NonStandardErpFederationsTest.java (hunk 1 above)
Comment (marcos-iov) [reply rskj-core/src/test/java/co/rsk/peg/federation/NonStandardErpFederationsTest.java]: updated

**Review by julia-zack — APPROVED** [review]
